### PR TITLE
Build versions widget in show page

### LINKED ISSUE #1038: Add version list to purl display
Based on the design for the version list: https://www.figma.com/proto/XjubCohES3rE8xSQpiNghq/Versioning?page-id=0%3A1&node-id=215-2&viewport=-12727%2C-3675%2C1&t=Q8XjDGjx2EhhCXEa-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=215%3A2

Add the version list to the right sidebar of the object display:

- [ ] Versions listed in descending numerical order
- [ ] Indicate which version is currently being displayed
- [ ] Show which version is currently being displayed
- [ ] Include "View" links to the display of each version

### LINKED ISSUE #1041: On version list, indicate withdrawn versions
Do not provide a link to view withdrawn versions, but they should still be present in the version list with the text "Withdrawn" instead of a view link.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #1038 
Fixes #1041

### CODE REVIEW

File: Procfile.dev [hunk 1]
```diff
@@ -1 +1 @@
-web: unset PORT && bin/rails server
\ No newline at end of file
+web: unset PORT && rdbg -O -n -c bin/rails -- server
```
Comment (mjgiarlo) [inline Procfile.dev:1]: Why is this here? Because it's useful to have a debugger when doing local dev. It's in a separate commit. See related README change.

File: README.md [hunk 2]
```diff
@@ -39,14 +39,30 @@ By default, local development will use production XML from PURL, so that you can
...
-  cocina: "<%= File.join(Rails.root, "document_cache") %>/%{druid_tree}/cocina.json"
+  public_xml: "%{root_path}/%{druid_tree}/public"
+  cocina: "%{root_path}/%{druid_tree}/cocina.json"
+  meta: "%{root_path}/%{druid_tree}/meta.json"
+  versioned:
+    public_xml: "%{root_path}/%{druid_tree}/%{druid}/versions/public.%{version_id}.xml"
+    cocina: "%{root_path}/%{druid_tree}/%{druid}/versions/cocina.%{version_id}.json"
+
+stacks:
+  version_manifest_path: "%{root_path}/%{druid_tree}/%{druid}/versions/versions.json"
```
Comment (mjgiarlo) [inline README.md:51]: These are needed for versioned layouts, so I added them.

File: app/models/purl_resource.rb [hunk 3]
```diff
@@ -32,6 +32,18 @@ def persisted?
...
   end
 
+  def versions
+    @versions = version_manifest&.fetch('versions', {})&.map do |version_id, version_attrs|
+      PurlVersion
+        .new(id:,
+             version_id:,
+             head: head_version == version_id.to_i,
+             updated_at: version_attrs.fetch('date', nil),
+             withdrawn: version_attrs.fetch('withdrawn', false))
```
Comment (mjgiarlo) [inline app/models/purl_resource.rb:42]: Might be worth a follow-up refactoring to add `PurlVersion.from_manifest` or some such?

File: app/models/purl_resource.rb [hunk 4]
```diff
@@ -68,14 +80,6 @@ def meta_json
 
   private
 
-  def versions
-    @versions = version_manifest&.fetch('versions', {})&.map do |version_id, _version_attrs|
-      PurlVersion.new(id:, version_id:, head: head_version == version_id.to_i).tap do |version|
-        raise PurlVersion::ObjectNotReady, id unless version.ready?
-      end
-    end
-  end
```
Comment (mjgiarlo) [inline app/models/purl_resource.rb:77]: Moved from private to public.

File: app/models/purl_version.rb [hunk 5]
```diff
@@ -18,11 +18,27 @@ def self.storage_root_path
     Settings.document_cache_root
   end
 
+  def head?
+    head
+  end
```
Comment (mjgiarlo) [inline app/models/purl_version.rb:23]: Moved up to be neared other attr-related methods.

File: app/models/purl_version.rb [hunk 6]
```diff
@@ -18,11 +18,27 @@ def self.storage_root_path
...
+
   # Coerce version IDs to integers
   def version_id=(value)
     @version_id = value.to_i
   end
 
+  def updated_at
+    @updated_at.presence || resource_retriever.updated_at
+  end
```
Comment (mjgiarlo) [inline app/models/purl_version.rb:36]: Get from version manifest else fallback to legacy behavior (mtime of public XML file)

File: app/models/purl_version.rb [hunk 7]
```diff
@@ -18,11 +18,27 @@ def self.storage_root_path
...
+
+  def updated_at=(value)
+    @updated_at = value&.to_datetime
+  end
```
Comment (mjgiarlo) [inline app/models/purl_version.rb:40]: Coerce to datetime.

File: app/models/purl_version.rb [hunk 8]
```diff
@@ -214,16 +226,6 @@ def cache_key
...
 
-    def updated_at
-      if public_xml_resource.respond_to? :updated_at
-        public_xml_resource.updated_at
-      elsif public_xml_resource.respond_to?(:header) && public_xml_resource.header[:last_modified].present?
-        last_modified_header_value
-      else
-        Time.zone.now
-      end
-    end
```
Comment (mjgiarlo) [inline app/models/purl_version.rb:225]: This was dead code, since `#updated_at` was delegated to the resource retriever.

**Review by jcoyne — APPROVED** [review]
